src/mlpack/core/data/load_impl.hpp

```diff
@@ -446,16 +409,16 @@ bool Load(const std::string& filename,
     if (transpose)
     {
       matrix.set_size(cols, rows);
-      info = DatasetInfo(cols);
+      info = DatasetMapper<PolicyType>(info.Policy(), cols);
```

Comment:
I think it's possible the problem I am thinking of existed in earlier versions of mlpack, but what happens if I do this:

```
arma::mat trainingSet, testSet;
DatasetInfo d;
data::Load("training_set.csv", trainingSet, d);
data::Load("test_set.csv", testSet, d);
```

Ideally the mappings from the first load should be preserved and used for the second load, but the lines here make me think that is not what is happening.  I think probably we should add a test for this situation, to ensure that mapping information from a previous load is not destroyed.